### PR TITLE
[ #188 ] Add retires to Jira functions to fix API rate limit related errors

### LINKED ISSUE #188: JIRA 401 Error raised when adding passed job notification
See [this build log](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-jboss-fuse-fuse-xpaas-qe-container-main-jboss-fuse-ocp4.16-lp-interop-jboss-fuse-interop-aws/1787362797902893056#1:build-log.txt%3A68).

This is the only instance observed at this time, but I will update here if I find others.

### PR DESCRIPTION
**Pull request by calebevans**
- Adds pyhelper-utils
- Adds use of retries in Jira related functions to avoide API rate limit errors

Closes #188

### CODE REVIEW

**Comment by msi-b1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * This repository does not support retest actions
</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: src/objects/jira_base.py [hunk 1]
```diff
@@ -5,8 +5,11 @@
 from jira import Issue
 from jira import JIRA
 from jira.exceptions import JIRAError
+from pyhelper_utils.general import ignore_exceptions
 from simple_logger.logger import get_logger
 
+LOGGER = get_logger(name=__name__)
+
```
Comment (dfrazzette) [inline src/objects/jira_base.py:12]:
```suggestion
LOGGER = get_logger(name=__name__)
```

Comment (dfrazzette) [reply src/objects/jira_base.py:12]: Delete extra line?

Comment (calebevans) [reply src/objects/jira_base.py:12]: This is due to linting. There should be two blank lines between a class and anything else.

**Pull request merged**